agent_config:
  agent_class: MosAgentTopo3D
  agent_type: local
  belief:
    init_params:
      num_samples: 3000
      occupancy_blow_up_res: 4
      occupancy_fill_height: true
      occupancy_height_thres: 0.2
      prior_from_occupancy: true
    visible_volume_params:
      num_rays: 150
      step_size: 0.2
      voxel_res: 2
  objects:
    Chair:
      class: Chair
      color:
      - 0.67
      - 0.61
      - 0.15
      - 0.8
      transition:
        class: sloop_object_search.oopomdp.StaticObjectTransitionModel
      viz_type: cube
  robot:
    action:
      policy:
        cost_scaling_factor: 1.0
      topo:
        3d_proj_2d:
          brush_size: 0.2
          floor_cut: 0.15
          inflation: 0.5
          layout_cut: 0.4
        debug: false
        num_nodes: 10
        pos_importance_thres: 0.01
        resample_thres: 0.4
        sep: 0.75
    detectors:
      Chair:
        class: sloop_object_search.oopomdp.FrustumVoxelAlphaBeta
        params:
          quality:
          - 10000000.0
          - 0.2
          sensor: gripper_camera
    id: robot0
    no_look: true
    reachable:
      max_height: 1.3
      min_height: 1.0
    sensors:
    - name: gripper_camera
      params:
        aspect_ratio: 0.7
        far: 1.5
        fov: 53
        near: 0.2
        occlusion_enabled: true
  search_region:
    3d:
      center_x: 0.0
      center_y: 0.0
      center_z: 0.0
      octree_size: 32
      region_size_x: 1.35
      region_size_y: 2.4
      region_size_z: 1.5
      res: 0.08
  targets:
  - Chair
planner_config:
  planner: pomdp_py.POUCT
  planner_params:
    exploration_const: 1000
    max_depth: 6
    num_sims: 400
    show_progress: true
task_config:
  max_steps: 200
world_state_config:
- color:
  - 0.05
  - 0.45
  - 0.72
  - 0.8
  name: table
  pose:
  - 0.43
  - 0
  - -0.49
  sizes:
  - 0.86
  - 1.16
  - 0.93
- color:
  - 0.95
  - 0.95
  - 0.88
  - 0.8
  name: xarm
  pose:
  - 0.8
  - 0.0
  - 0.475
  sizes:
  - 0.4
  - 0.4
  - 1.0
- color:
  - 0.18
  - 0.18
  - 0.18
  - 0.8
  name: chair
  pose:
  - 0.05
  - 1.37
  - -0.54
  sizes:
  - 0.4
  - 0.4
  - 0.83
- color:
  - 0.5
  - 0.8
  - 0.9
  - 0.8
  name: arm_origin
  pose:
  - 0.0
  - 0.0
  - 0.0
  sizes:
  - 0.17
  - 0.17
  - 0.25
